R/reports.R

```diff
@@ -0,0 +1,23 @@
+get_report_parameters <- function(name, ref, root) {
+  path <- get_orderly_script_path(name, ref, root)
+  sha <- git_ref_to_sha(ref, repo = root, check = TRUE)
+  contents <- git_run(
+    c("show", sprintf("%s:%s", sha, path)), repo = root, check = TRUE
+  )$output
+  exprs <- parse(text = contents)
+  orderly2::orderly_parse(exprs = exprs, filename = basename(path))$parameters
+}
+
+
+get_orderly_script_path <- function(name, ref, root) {
+  contents <- gert::git_ls(root, ref = ref)
+  re <- sprintf("^src/%s/(%s|orderly)\\.R$", name, name)
+  matches <- grep(re, contents$path, value = TRUE, perl = TRUE)
+  if (length(matches) > 1) {
```

Comment:
should we care about the case where they have none? or assumed to have at least one at this point?

Reply:
API should only call this for directories which have an `orderly.R` script. So this shouldn't be triggered with matches 0. That said, better to not rely on that knowledge as the error message raised if there is no file is quite ugly. So I've expanded this check to test if `!= 0`